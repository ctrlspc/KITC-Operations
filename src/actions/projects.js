import database from '../firebase/firebase';

export const setProjects = (projects) => ({
  type: 'SET_PROJECTS',
  projects
});

export const startSetProjects = () => {
  return (dispatch) => {
    return database.ref(`projects`)
      .once('value')
      .then((snapshot) => {
        const projects = [];

        snapshot.forEach(childsnapshot => {
          projects.push({
            id:childsnapshot.key,
            ...childsnapshot.val()
          });
        });        
        dispatch(setProjects(projects));
      });
  };
};

export const addProject = (project) => ({
  type: 'ADD_PROJECT',
  project
});

export const startAddProject = ({title, description, projectManager, projectType}) => {
 return (dispatch) => {
    const project = {title, description, projectManager, projectType};

    return database.ref(`projects`)
      .push(project)
      .then((ref) => {
        dispatch(addProject({
          id:ref.key,
          ...project
        }));

        return Promise.resolve(ref.key);
      });
  };
};

export const updateProject = (id, updates) => ({
  type: 'UPDATE_PROJECT',
  id, 
  updates
});

export const startUpdateProject = (id, updates) => {
  return (dispatch) => {
    return database.ref(`projects/${id}`)
      .update(updates)
      .then((ref) => {
        dispatch(updateProject(id, updates));
      });
    };
};
