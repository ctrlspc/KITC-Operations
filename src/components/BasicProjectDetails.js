import React from 'react';

export default ({title, description, projectManager, projectType}) => (
  <div>
    <h3>{title}</h3>
    <p>{description}</p>
    <p><span>Project Manager:</span> {projectManager.name}</p>
  </div>
);