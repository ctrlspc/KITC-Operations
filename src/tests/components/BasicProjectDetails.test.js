import React from 'react';
import { shallow } from 'enzyme';
import BasicProjectDetails from '../../components/BasicProjectDetails';
import projects from '../fixtures/projects';

test('should render a NotFoundPageComponent', () => {
  const wrapper = shallow(<BasicProjectDetails {...projects[0]}/>);
  expect(wrapper).toMatchSnapshot();
});