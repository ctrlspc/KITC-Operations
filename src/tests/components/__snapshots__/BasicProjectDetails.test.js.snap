// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a NotFoundPageComponent 1`] = `
<div>
  <h3>
    first test project
  </h3>
  <p>
    testing some great stuff
  </p>
  <p>
    <span>
      Project Manager:
    </span>
     
    Mr Test
  </p>
</div>
`;
